contracts/crosschain/ERC7802Bridge.sol

```diff
@@ -0,0 +1,256 @@
+        _requireOwned(uint256(bridgeId));
+        return _bridges[bridgeId].gateway[chain];
+    }
+
+    function getBridgeRemote(bytes32 bridgeId, bytes memory chain) public view returns (bytes memory) {
+        _requireOwned(uint256(bridgeId));
+        return _bridges[bridgeId].remote[chain];
+    }
+
+    // ┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+    // │                                     Bridge creation and administration                                      │
+    // └─────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+    struct Foreign {
+        bytes32 id;
+        address gateway;
+        bytes remote;
+    }
+
+    function createBridge(
+        address token,
+        address admin,
+        bool isCustodial,
+        Foreign[] calldata foreign
+    ) public returns (bytes32) {
+        bytes32[] memory ids = new bytes32[](foreign.length + 1);
+        bytes32[] memory links = new bytes32[](foreign.length);
+        for (uint256 i = 0; i < foreign.length; ++i) {
+            require(foreign[i].gateway != address(0) && foreign[i].remote.length > 0);
+            ids[i] = foreign[i].id;
+            links[i] = keccak256(
+                abi.encode(InteroperableAddress.formatEvmV1(block.chainid, foreign[i].gateway), foreign[i].remote)
+            );
+        }
+        ids[foreign.length] = keccak256(
+            abi.encode(
+                InteroperableAddress.formatEvmV1(block.chainid, token), // bytes token
+                bytes32(bytes20(admin)) | bytes32(SafeCast.toUint(isCustodial)), // bytes32 tokenOptions
+                Arrays.sort(links)
+            )
+        );
+
+        bytes32 bridgeId = keccak256(abi.encodePacked(Arrays.sort(ids)));
+
+        // Should we check for collision. I don't think that is necessary
+        BridgeMetadata storage details = _bridges[bridgeId];
+        details.token = token;
+        details.isCustodial = isCustodial;
+
+        _safeMint(admin == address(0) ? address(1) : admin, uint256(bridgeId));
+
+        for (uint256 i = 0; i < foreign.length; ++i) {
+            (bytes2 chainType, bytes memory chainReference, ) = foreign[i].remote.parseV1();
+            bytes memory chain = InteroperableAddress.formatV1(chainType, chainReference, "");
+            require(details.gateway[chain] == address(0));
+            details.gateway[chain] = foreign[i].gateway;
+            details.remote[chain] = foreign[i].remote;
+
+            emit BridgeLinkSet(bridgeId, foreign[i].gateway, foreign[i].remote);
+        }
+
+        return bridgeId;
+    }
+
+    function setPaused(bytes32 bridgeId, bool isPaused) public bridgeAdminRestricted(bridgeId) {
+        _bridges[bridgeId].isPaused = isPaused;
+        emit BridgePaused(bridgeId, isPaused);
+    }
+
+    function updateGateway(
```

Comment:
currently, it is possible to add new chains to a bridgeId. This should not be allowed unless there is a mechanism to create the bridgeId on that new chain that is not part of the initial config.